Sources/NFCPassportReader/PassportReader.swift

```diff
@@ -199,10 +227,20 @@ extension PassportReader : NFCTagReaderSessionDelegate {
             } catch let error as NFCPassportReaderError {
                 let errorMessage = NFCViewDisplayMessage.error(error)
                 self.invalidateSession(errorMessage: errorMessage, error: error)
-            } catch let error {
+            } catch {
                 Logger.passportReader.debug( "tagReaderSession:failed to connect to tag - \(error.localizedDescription)" )
-                let errorMessage = NFCViewDisplayMessage.error(NFCPassportReaderError.ConnectionError)
-                self.invalidateSession(errorMessage: errorMessage, error: NFCPassportReaderError.Unknown(error))
+
+                if let nfcError = error as? NFCReaderError {
+                    // .readerTransceiveErrorTagResponseError is thrown when a "connection lost" scenario is forced by moving the phone away from the NFC chip
+                    // .readerTransceiveErrorTagConnectionLost is never thrown for this scenario, but added for the sake of completeness
+                    if nfcError.errorCode == NFCReaderError.readerTransceiveErrorTagResponseError.rawValue || nfcError.errorCode == NFCReaderError.readerTransceiveErrorTagConnectionLost.rawValue {
```

Comment:
We dont need `else` for this `if`?

Reply:
```suggestion
                    if let nfcError = error as? NFCReaderError,
                       nfcError.errorCode == NFCReaderError.readerTransceiveErrorTagResponseError.rawValue ||
                        nfcError.errorCode == NFCReaderError.readerTransceiveErrorTagConnectionLost.rawValue
```

Good point, I suggest combining the two `if` into a single `if`, such that the `else` is handled by the already existing `else` case of an Unknown Error.